### DISCUSSION TITLE
Suddenly, can't upload via serial to an "Adafruit Feather RP2040 with USB Type A Host" board

### DISCUSSION THREAD

**Discussion by heliophagus** [earlephilhower/arduino-pico#2233] [General]
Hi Earle - thank you for the amazing work you do!

I've hit a roadblock with an "Adafruit Feather RP2040 with USB Type A Host" board, using Arduino IDE 2.3.2 & version 3.9.2 of your Arduino core. After using the board successfully for a few days, it suddenly disabled uploading via serial. It errors with "No drive to deploy. Failed uploading: uploading error: exit status 1." The TOOLS/PORT option is disabled.

If I put the board into boot mode, I can apparently upload programs - the drive appears after entering boot mode & the IDE apparently uploads to it, after which the drive disappears as usual. But then the Uploading progress bar lingers for about 30 seconds, after which the IDE errors with "Port monitor error: command 'open' failed: Serial port not found" & now the TOOLS/PORT option is disabled again. So I can't debug via serial, which I rely on at this stage of the project. Unfortunately the board is out of stock, so I'd like to get this one working. I tried erasing the board's flash memory (no effect) & getting input from the Adafruit forums, but received nothing helpful - quite the opposite. So I can't tell if this is a board fault, or a weird edge case that can somehow be resolved, or must remain a mystery.

BTW, the IDE & RP2040 core work perfectly with other RP2040 boards.

If you have any helpful suggestions, I'm all ears!

**Comment by earlephilhower** [comment]
Err, that's not actually a listed board type in the menus.  I assume you're using the plain Adafruit Feather RP2040 board type instead, yes?

Anyway, looking at the board who drives the USB host?  You can't do USB slave (i.e. serial port) and USB master (i.e. that type-A port) at the same time, it's not possible with the HW onboard.

There have been no changes to the core related to this, so you'll need to dig into what's changed on your end.  Try something simple like the blink example with nothing connected to the USBA port.  If you see it blinking but can't see the serial port, then your OS has a problem but the Pico is fine.  If blink.ino doesn't blink and you don't have a serial port, then you may have some HW issue on the board.

The USB port is SW driven from the Pico, so if the app is crashed or hung then you won't see a port, either.

If you're running under Windows, double check you have followed the Zadig installation or else Windows will insist the device doesn't exist since it doesn't have an `.inf` or related about it.

Sometimes Windows or Linux itself gets persnickety about USB ports and you can try a different port and cable to see if that helps change things.

**Reply by heliophagus** [reply]
Thanks. This is the board: https://www.adafruit.com/product/5723 - it uses PIO to bit-bang the USB A interface (see https://github.com/sekigon-gonnoc/Pico-PIO-USB ) & it worked well. It was happily writing .csv logs to a thumb drive during several days of development, using serial upload from the IDE as usual, before it lost its mind. I can upload blink to it if I kick it into boot mode, then upload via the IDE - the board blinks, but no serial port appears & it errors out with a port monitor error. I installed Zadig but it made no difference. Maybe this will just remain a mystery.

**Reply by heliophagus** [reply]
(BTW I used the same cable with other boards with no issues.)

**Reply by earlephilhower** [reply]
If the blinker blinks then there is an OS issue WRT the USB serial port.  You might fake out Windows by using the `generic` board instead of the Adafruit one in the menus, and then using Zadig to add the generic board's serial port.

**Reply by heliophagus** [reply]
Thanks for the suggestion. Tried everything but the board's serial port appears to be kaput. Blink no longer loads, even in boot mode. Other boards work well. So evidently a board issue after all. I wonder if these issues may explain why this board has been "out of stock" at Adafruit for so long, with no ETA expected (I wrote them & asked), plus their silence on this subject at their support forum. Ah well, will need to roll my own board!

**Reply by earlephilhower** [reply]
Very wierd.  You can also use a Picoprobe and the SWD debug port to upload apps to the board.  If you don't need the serial port for anything other than programming it's not a big deal.

I have a board where the bootsel button failed and if the app crashes I need to use this port to upload a new sketch.  It just happens with these micro buttons.
